### PR TITLE
Add some working examples

### PR DESCRIPTION
**Pull request by prologic**
Ported some of these from monkey-lang 😅 More to come in another PR

### CODE REVIEW

File: examples/fib.tau [hunk 1]
```diff
@@ -0,0 +1,11 @@
+fib = fn(x) {
+  return if x == 0 {
+    0
+  } else if x == 1 {
+    1
+  } else {
+    fib(x-1) + fib(x-2)
+  }
+}
```
Comment (prologic) [inline examples/fib.tau:9]: Oooh I didn't know you could do this in Tau! Very nice 👌

Comment (NicoNex) [reply examples/fib.tau:9]: I probably need to write more explicative READMEs 😆.

Comment (prologic) [reply examples/fib.tau:9]: The README is fine, but I _think_ you should highlight this point a bit more, maybe even make it it's own section 😅

Comment (NicoNex) [reply examples/fib.tau:9]: Thinking about it, with Tau you can also write the function like this, without the explicit return:
```python
fib = fn(x) {
  if x == 0 {
    0
  } else if x == 1 {
    1
  } else {
    fib(x-1) + fib(x-2)
  }
}
```

Comment (prologic) [reply examples/fib.tau:9]: I've got a few other variations I want to try in Tau and port from Monkey Lang just for the fun 😅

**Pull request merged**